python/lsst/pex/config/config.py

```diff
@@ -37,9 +37,31 @@
 import shutil
 import warnings
 
+# if YAML is not available that's fine and we simply don't register
+# the yaml representer since we know it won't be used.
+try:
+    import yaml
+    # doImport is needed to preimport the Config class on yaml load
+    from lsst.utils import doImport
+except ImportError:
+    yaml = None
+    YamlLoaders = ()
+    doImport = None
+
 from .comparison import getComparisonName, compareScalars, compareConfigs
 from .callStack import getStackFrame, getCallStack
 
+if yaml:
+    try:
+        # CLoader is not always available
+        from yaml import CLoader
+    except ImportError:
+        CLoader = None
+
+    YamlLoaders = (yaml.Loader, yaml.FullLoader, yaml.SafeLoader, yaml.UnsafeLoader)
```

Comment:
Are we sure that `FullLoader` is always available (requires relatively recent PyYAML)?  Should we be adding to all of these loaders, or just `SafeLoader` so that we don't (passively) encourage people to do unsafe things?

Reply:
We are requiring pyyaml>5 so we should be fine. I have tried just registering SafeLoader but that really does mean that no other loader works. Seems weird to me since if you are allowing it to work in safe mode then surely it should work in unsafe mode.